### DISCUSSION TITLE
What is the correct prop-type for the service obtained from useInterpret hook?

### DISCUSSION THREAD

**Discussion by sowizz** [statelyai/xstate#3291] [Q&A] [answered]
Hello, 

I'm struggling with defining prop type for the `service` obtained from `useInterpret` hook.
My parent component passes `service` to it's children, like in one of the examples in the documentation:
```reactjs
import { useSelector } from '@xstate/react';

// tip: optimize selectors by defining them externally when possible
const selectCount = (state) => state.context.count;

const App = ({ service }) => {
  const count = useSelector(service, selectCount);

  // ...
};
```
I've tried following approach:

```reactjs
import { Interpreter } from 'xstate';

//...

App.propTypes = {
  service: PropTypes.objectOf(Interpreter).isRequired,
};
```
, but it's throwing following warning into the console:

> Warning: Failed prop type: Cannot set properties of undefined (setting 'machine')

Any idea how do I define the PropType for the `service` prop correctly?

**Comment by Andarist** [comment, marked as answer]
This should do the trick
```js
App.propTypes = {
  service: PropTypes.instanceOf(Interpreter).isRequired
};
```

https://codesandbox.io/s/proud-hooks-uq1bxi?file=/src/App.js:402-480

**Reply by sowizz** [reply]
I've tried to use `instanceOf` too and it fails with following warning:

> Warning: Failed prop type: Invalid prop `service` of type `Interpreter` supplied to `TrendContent`, expected instance of `Interpreter`.

**Reply by Andarist** [reply]
Well, you will have to share a runnable repro case for this issue then because the very basic example that I've prepared works just alright. So something else is at play here - but I can't just guess what it is.

**Reply by sowizz** [reply]
Hi, you were right, it works. I had Xstate installed using npm aliases - to have two different versions and migrate to new one step, by step - and it caused this issue. Thanks!
